sds/src/scoped_ruleset/mod.rs

```diff
@@ -199,22 +215,52 @@ where
         // Sanitize the segment and push it
         self.sanitized_segments_until_node.push(segment.sanitize());
 
+        let true_positive_rules_count = if self.should_keywords_match_event_paths {
+            let mut current_sanitized_path = self
+                .sanitized_segments_until_node
+                .iter()
+                .filter_map(|sanitized_segment| {
+                    sanitized_segment
+                        .as_ref()
+                        .map_or(None::<Cow<str>>, |x| Some(x.clone()))
+                })
+                .fold(String::new(), |mut a, b| {
+                    a.reserve(b.len() + 1);
+                    a.push_str(&*b);
+                    a.push(UNIFIED_LINK_CHAR);
+                    a
+                });
```

Comment:
This technique is more efficient than collect: https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=bacb488f2c7df299c244403556cd9b25